datamodel/samples.d/generate_timestampdefinitions.py

```diff
@@ -0,0 +1,1188 @@
+    )
+
+    # ETS-Mooring (Inbound) UC 19 - 24
+    xty_service_timestamps(
+        EST_REQ_PLN,  # ATS-Cargo Ops comes later (different port call part)
+        ['MOOR'],
+        [NULL_VALUE, 'INBD'],
+        "Services Planning",
+        'jit1_2',
+        include_phase_in_name=True,
+        is_cancelable=True,
+        need_vessel_position_for=EST_PLN,
+        need_event_location_for=REQ_PLN_ACT,
+    )
+
+    # Bunkering UC 25 - 30
+    xty_service_timestamps(
+        EST_REQ_PLN,
+        ['BUNK'],
+        [NULL_VALUE, 'ALGS'],
+        "Services Planning",
+        'jit1_2',
+        include_phase_in_name=False,
+        is_cancelable=True,
+        need_event_location_for=ALL_EVENT_CLASSIFIER_CODES,
+    )
+
+    # ETA PBP UC 31-33 + 35
+    generic_xty_timestamps(
+        PUBLISHER_PATTERN_CA2ATH,
+        ALL_EVENT_CLASSIFIER_CODES,
+        ['ARRI'],
+        ['PBPL'],
+        [NULL_VALUE, 'INBD'],
+        "Pilot Boarding Place Arrival Planning And Execution, Berth Arrival Execution",
+        'jit1_0',
+        need_event_location_for=ALL_EVENT_CLASSIFIER_CODES,
+    )
+
+    # EOSP UC 34
+    generate_special_timestamp(
+        'ESOP',
+        PUBLISHER_PATTERN_CA2ATH,
+        'ARRI',
+        'BRTH',
+        'INBD',
+        'Pilot Boarding Place Arrival Planning And Execution, Berth Arrival Execution',
+        'jit1_1',
+        need_vessel_position=True,  # FIXME: IFS says Optional, we do not support "required" or "absent".
+    )
+
+    # ATS-Pilotage (Inbound) UC 36 + 40
+    xty_service_timestamps(
+        ACT,
+        ['PILO'],
+        [NULL_VALUE, 'INBD'],
+        "Pilot Boarding Place Arrival Planning And Execution, Berth Arrival Execution",
+        'jit1_0',
+        include_phase_in_name=True,
+        is_cancelable=True,
+        need_vessel_position_for=ACT,
+        need_event_location_for=ACT,
+    )
+
+    # ATS-Towage (Inbound) UC 37 + 38
+    xty_service_timestamps(
+        ACT,
+        ['TOWG'],
+        [NULL_VALUE, 'INBD'],
+        "Pilot Boarding Place Arrival Planning And Execution, Berth Arrival Execution",
+        'jit1_0',
+        include_phase_in_name=True,
+        is_cancelable=True,
+        need_vessel_position_for=ACT,
+        need_event_location_for=ACT,
+    )
+
+    # ATS-Mooring (Inbound) UC 39 + 44
+    xty_service_timestamps(
+        ACT,
+        ['MOOR'],
+        [NULL_VALUE, 'INBD'],
+        "Pilot Boarding Place Arrival Planning And Execution, Berth Arrival Execution",
+        'jit1_0',
+        include_phase_in_name=True,
+        is_cancelable=True,
+        need_event_location_for=ACT,
+    )
+
+    #ATA-Berth UC 41
+    generate_special_timestamp(
+        'ATA-Berth',
+        as_publisher_patterns(['TR'], ['ATH']),
+        'ARRI',
+        'BRTH',
+        'ALGS',
+        "Pilot Boarding Place Arrival Planning And Execution, Berth Arrival Execution",
+        'jit1_0',
+        need_vessel_position=False,
+    )
```

Comment:
Could have sworn that one was a `TR` -> `CA` but you are indeed right.

In practice, I think the `CA` (et al.) is allowed to publish this one also, so I would hadd CA -> ATH in addition (not sure how to do that in practice off-hand). To be confirmed with Charles.